packages/devtools_app/lib/src/screens/debugger/debugger_controller.dart

```diff
@@ -307,38 +308,38 @@ class DebuggerController extends DisposableController
   ValueListenable<List<StackFrameAndSourcePosition>>
       get stackFramesWithLocation => _stackFramesWithLocation;
 
-  final _selectedStackFrame = ValueNotifier<StackFrameAndSourcePosition>(null);
+  final _selectedStackFrame = ValueNotifier<StackFrameAndSourcePosition?>(null);
 
-  ValueListenable<StackFrameAndSourcePosition> get selectedStackFrame =>
+  ValueListenable<StackFrameAndSourcePosition?> get selectedStackFrame =>
       _selectedStackFrame;
 
-  Frame get frameForEval =>
+  Frame? get frameForEval =>
       _selectedStackFrame.value?.frame ??
-      _stackFramesWithLocation.value?.safeFirst?.frame;
+      _stackFramesWithLocation.value.safeFirst?.frame;
 
   final _variables = ValueNotifier<List<DartObjectNode>>([]);
 
   ValueListenable<List<DartObjectNode>> get variables => _variables;
 
-  final _breakpoints = ValueNotifier<List<Breakpoint>>([]);
+  final _breakpoints = ValueNotifier<List<Breakpoint>?>([]);
```

Comment:
this list can be non-nullable. We are using the empty list as the base state and not null.